src/qt/optionsmodel.cpp

```diff
@@ -381,6 +386,27 @@ bool OptionsModel::setData(const QModelIndex & index, const QVariant & value, in
                 setRestartRequired(true);
             }
             break;
+        case AddressTypeLegacy:
+            if (value.toBool()) {
+                settings.setValue("strAddressType", QString::fromStdString(FormatOutputType(OUTPUT_TYPE_LEGACY)));
+                g_address_type = OUTPUT_TYPE_LEGACY;
```

Comment:
Not sure, but I guess changing `g_address_type` and `g_change_type ` needs locking (concurrency), otherwise a race would be possible (unlikely, though possible).

Reply:
Guard with cs_main? Most accesses already held the lock. Fix the remaining.